[CI] DatabaseNodeServiceIT testGzippedDatabase failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #4199 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/7b4thv7kk6vtu)
- [elasticsearch-periodic #4226 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/u54naprla75qg)
- [elasticsearch-periodic-platform-support #4199 / almalinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ayaqjtotauyui)
- [elasticsearch-periodic-platform-support #4199 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/sqlizezhzxoii)
- [elasticsearch-periodic-platform-support #4195 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/vgkhimc5vjmfw)
- [elasticsearch-periodic-platform-support #4195 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/upvochxkzalc6)
- [elasticsearch-periodic-platform-support #4195 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/zg7upagwrlsng)
- [elasticsearch-periodic-platform-support #4195 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/p7oafeih6lzx2)
- [elasticsearch-periodic #4222 / openjdk21_checkpart1_java-fips-matrix](https://gradle-enterprise.elastic.co/s/ix5tc3ojhn2d6)
- [elasticsearch-periodic-platform-support #4191 / ubuntu-2004-aarch64_checkpart1_platform-support-arm](https://gradle-enterprise.elastic.co/s/esw5japd45jlu)

**Reproduction Line:**
```
./gradlew ":modules:ingest-geoip:internalClusterTest" --tests "org.elasticsearch.ingest.geoip.DatabaseNodeServiceIT.testGzippedDatabase" -Dtests.seed=FE57913218ADB1C7 -Dtests.locale=id -Dtests.timezone=America/Yakutat -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testGzippedDatabase'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.ingest.geoip.DatabaseNodeServiceIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: expected null, but was:<org.elasticsearch.ingest.geoip.DatabaseReaderLazyLoader@330850e6>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testGzippedDatabase
- [main] 3 consecutive failures in step ubuntu-2004-aarch64_checkpart1_platform-support-arm
- [main] 2 consecutive failures in step rhel-7_platform-support-unix
- [main] 11 failures in test testGzippedDatabase (15.5% fail rate in 71 executions)
- [main] 3 failures in step ubuntu-2004-aarch64_checkpart1_platform-support-arm (100.0% fail rate in 3 executions)
- [main] 2 failures in step rhel-7_platform-support-unix (66.7% fail rate in 3 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 3 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/DatabaseNodeServiceIT.java
+++ b/modules/ingest-geoip/src/internalClusterTest/java/org/elasticsearch/ingest/geoip/DatabaseNodeServiceIT.java
@@ -44,15 +44,17 @@ public class DatabaseNodeServiceIT extends AbstractGeoIpIT {
      * This test makes sure that if we index an ordinary mmdb file into the .geoip_databases index, it is correctly handled upon retrieval.
      */
     public void testNonGzippedDatabase() throws Exception {
-        String databaseName = "GeoLite2-Country";
-        String databaseFileName = databaseName + ".mmdb";
+        String databaseType = "GeoLite2-Country";
+        String databaseFileName = databaseType + ".mmdb";
+        // making the dabase name unique so we know we're not using another one:
+        String databaseName = randomAlphaOfLength(20) + "-" + databaseFileName;
         byte[] mmdbBytes = getBytesForFile(databaseFileName);
         final DatabaseNodeService databaseNodeService = internalCluster().getInstance(DatabaseNodeService.class);
-        assertNull(databaseNodeService.getDatabase(databaseFileName));
-        int numChunks = indexData(databaseFileName, mmdbBytes);
-        retrieveDatabase(databaseNodeService, databaseFileName, mmdbBytes, numChunks);
-        assertBusy(() -> assertNotNull(databaseNodeService.getDatabase(databaseFileName)));
-        assertValidDatabase(databaseNodeService, databaseFileName, databaseName);
+        assertNull(databaseNodeService.getDatabase(databaseName));
+        int numChunks = indexData(databaseName, mmdbBytes);
+        retrieveDatabase(databaseNodeService, databaseName, mmdbBytes, numChunks);
+        assertBusy(() -> assertNotNull(databaseNodeService.getDatabase(databaseName)));
+        assertValidDatabase(databaseNodeService, databaseName, databaseType);
     }
 
     /*
@@ -60,16 +62,18 @@ public class DatabaseNodeServiceIT extends AbstractGeoIpIT {
      * handled upon retrieval.
      */
     public void testGzippedDatabase() throws Exception {
-        String databaseName = "GeoLite2-Country";
-        String databaseFileName = databaseName + ".mmdb";
+        String databaseType = "GeoLite2-Country";
+        String databaseFileName = databaseType + ".mmdb";
+        // making the dabase name unique so we know we're not using another one:
+        String databaseName = randomAlphaOfLength(20) + "-" + databaseFileName;
         byte[] mmdbBytes = getBytesForFile(databaseFileName);
-        byte[] gzipBytes = gzipFileBytes(databaseFileName, mmdbBytes);
+        byte[] gzipBytes = gzipFileBytes(databaseName, mmdbBytes);
         final DatabaseNodeService databaseNodeService = internalCluster().getInstance(DatabaseNodeService.class);
-        assertNull(databaseNodeService.getDatabase(databaseFileName));
-        int numChunks = indexData(databaseFileName, gzipBytes);
-        retrieveDatabase(databaseNodeService, databaseFileName, gzipBytes, numChunks);
-        assertBusy(() -> assertNotNull(databaseNodeService.getDatabase(databaseFileName)));
-        assertValidDatabase(databaseNodeService, databaseFileName, databaseName);
+        assertNull(databaseNodeService.getDatabase(databaseName));
+        int numChunks = indexData(databaseName, gzipBytes);
+        retrieveDatabase(databaseNodeService, databaseName, gzipBytes, numChunks);
+        assertBusy(() -> assertNotNull(databaseNodeService.getDatabase(databaseName)));
+        assertValidDatabase(databaseNodeService, databaseName, databaseType);
     }
 
     /*

